tests/compilers/test_ast.py

```diff
@@ -664,3 +668,8 @@ def test_ast_good_yield_from():
 def test_ast_bad_yield_from():
     "Make sure AST can't compile invalid yield-from"
     cant_compile("(yield-from)")
+
+
+def test_eval_generator_with_return():
+    """Ensure generators with a return statement works."""
+    can_eval("(fn [] (yield 1) (yield 2) (return))")
```

Comment:
Has to be evaluated as the outputted AST is fine. Python rejects it only on import.